File: June-LeetCoding-Challenge/29-Unique-Paths/Unique-Paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<vector<int>> dp(n, vector<int>(m, 0));
        for (int i = 0; i < m; i++) dp[0][i] = 1;
        for (int i = 0; i < n; i++) dp[i][0] = 1;
        
        for (int i = 1; i < n; i++)
            for (int j = 1; j < m; j++)
                dp[i][j] = dp[i-1][j] + dp[i][j-1];
        return dp[n-1][m-1];
    }
};

File: June-LeetCoding-Challenge/29-Unique-Paths/Unique-Paths.kt
package dynamic_programming

class UniquePathsKotlin62 {
    fun uniquePaths(m: Int, n: Int): Int {
        val dp = Array(m + 1) { IntArray(n + 1) }
        dp[0][1] = 1
        for (i in 1..m) {
            for (j in 1..n) {
                dp[i][j] = dp[i - 1][j] + dp[i][j - 1]
            }
        }
        return dp[m][n]
    }
}

fun main() {
    val solution = UniquePathsKotlin62()
    // 3
    println(solution.uniquePaths(3, 2))
    // 28
    println(solution.uniquePaths(7, 3))
}